quinn/src/runtime/async_std.rs

```diff
@@ -23,10 +24,11 @@ impl Runtime for AsyncStdRuntime {
         async_std::task::spawn(future);
     }
 
-    fn wrap_udp_socket(&self, sock: std::net::UdpSocket) -> io::Result<Box<dyn AsyncUdpSocket>> {
+    fn wrap_udp_socket(&self, sock: std::net::UdpSocket) -> io::Result<Arc<dyn AsyncUdpSocket>> {
         udp::UdpSocketState::configure((&sock).into())?;
-        Ok(Box::new(UdpSocket {
+        Ok(Arc::new(UdpSocket {
```

Comment:
Style suggestion (unrelated to the change): maybe `.into()` instead of `Arc::new()`?

Reply:
That doesn't work, because we're relying on unsizing to coerce `Arc<UdpSocket>` to `Arc<dyn AsyncUdpSocket>`, and the latter doesn't implement `From`. I like the explicitness, anyway.